### ISSUE TITLE
Composer ignores caret

### ISSUE THREAD

**Issue by martixy** [composer/composer#5151]
With the following composer.json:
(After the command.)
```json
{
    "name": "[...]/",
    "license": "proprietary",
    "type": "project",
    "autoload": {
        "psr-4": {
            "": "src/"
        },
        "classmap": [
            "app/AppKernel.php",
            "app/AppCache.php"
        ]
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "require": {
        "php": ">=5.5.9",
        "symfony/symfony": "3.0.*",
        "doctrine/orm": "^2.5",
        "doctrine/doctrine-bundle": "^1.6",
        "doctrine/doctrine-cache-bundle": "^1.2",
        "symfony/swiftmailer-bundle": "^2.3",
        "symfony/monolog-bundle": "^2.8",
        "sensio/distribution-bundle": "^5.0",
        "sensio/framework-extra-bundle": "^3.0.2",
        "incenteev/composer-parameter-handler": "^2.0",
        "javiereguiluz/easyadmin-bundle": "^1.12",
        "doctrine/doctrine-migrations-bundle": "1.1"
    },
    "require-dev": {
        "sensio/generator-bundle": "^3.0",
        "symfony/phpunit-bridge": "^2.7"
    },
    "scripts": {
        "post-install-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ],
        "post-update-cmd": [
            "Incenteev\\ParameterHandler\\ScriptHandler::buildParameters",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installRequirementsFile",
            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::prepareDeploymentTarget"
        ]
    },
    "extra": {
        "symfony-app-dir": "app",
        "symfony-bin-dir": "bin",
        "symfony-var-dir": "var",
        "symfony-web-dir": "web",
        "symfony-tests-dir": "tests",
        "symfony-assets-install": "relative",
        "incenteev-parameters": {
            "file": "app/config/parameters.yml"
        }
    }
}
```

When I run this command:

```
composer require doctrine/doctrine-migrations-bundle "^1.1" -vvv
```

I get this output:

```
D:\WebDev\[...] [dev +1 ~1 -0 !]> composer require doctrine/doctrine-migrations-bundle "^1.1" -vvv
Reading ./composer.json
Loading config file ./composer.json
Checking CA file C:\Users\Martixy\AppData\Local\Temp\composer-cacert-39eaf16e26c08cc90d728865f1c60406ee5d864d5963f72ede2a1cbb127f95ca.pem
Executing command (D:\WebDev\[...]): git branch --no-color --no-abbrev -v
Failed to initialize global composer: Composer could not find the config file: C:/Users/Martixy/AppData/Roaming/Composer/composer.json
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Reading D:\WebDev\[...]/vendor/composer/installed.json
./composer.json has been updated
Reading ./composer.json
Loading config file ./composer.json
Executing command (D:\WebDev\[...]): git branch --no-color --no-abbrev -v
Failed to initialize global composer: Composer could not find the config file: C:/Users/Martixy/AppData/Roaming/Composer/composer.json
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Reading D:\WebDev\[...]/vendor/composer/installed.json
Reading ./composer.lock
Loading composer repositories with package information
Downloading https://packagist.org/packages.json
Writing C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/packages.json into cache
Updating dependencies (including require-dev)
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/p-provider-2013.json from cache
Downloading http://packagist.org/p/provider-2014%24c16f80a6fc9a8e03fcc3796b12bd12c93cc3984edc9b8b9dc79fd33573678941.json
Writing C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/p-provider-2014.json into cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/p-provider-2015.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/p-provider-2015-07.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/p-provider-2015-10.json from cache
Downloading http://packagist.org/p/provider-2016-01%24746db5396a4906cad2d9ab4452d10ff265b0de58a3dc03bb0e58086a19bd2d9c.json
Writing C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/p-provider-2016-01.json into cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/p-provider-archived.json from cache
Downloading http://packagist.org/p/provider-latest%2444b7c28fa63d82aa05f770962cc1367e48fda47b7b0cf2db1d5df206f937fbae.json
Writing C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/p-provider-latest.json into cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$lexer.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$annotations.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$collections.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$inflector.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$cache.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$common.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$dbal.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-twig$twig.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$polyfill-util.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-paragonie$random-compat.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$polyfill-php70.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$polyfill-php56.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$polyfill-mbstring.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-psr$log.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$polyfill-intl-icu.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$symfony.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$intl.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$icu.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$polyfill-php54.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$doctrine-bridge.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$doctrine-cache-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$instantiator.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$yaml.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-incenteev$composer-parameter-handler.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-twig$extensions.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$dependency-injection.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$framework-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-sensio$framework-extra-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$config.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$event-dispatcher.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$http-kernel.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$filesystem.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$routing.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$stopwatch.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$templating.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$translation.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$http-foundation.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$finder.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$security-core.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$security-csrf.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$asset.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$class-loader.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$debug.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$polyfill-php55.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$polyfill-apcu.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-ircmaxell$password-compat.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$console.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-sensiolabs$security-checker.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$process.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-sensio$distribution-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-pagerfanta$pagerfanta.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$orm.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-jdorn$sql-formatter.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$doctrine-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$security-acl.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$form.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$property-access.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$twig-bridge.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-javiereguiluz$easyadmin-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$options-resolver.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-monolog$monolog.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$monolog-bridge.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$monolog-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-swiftmailer$swiftmailer.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$swiftmailer-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-sensio$generator-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$phpunit-bridge.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$swiftmailer-bridge.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$doctrine-migrations-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-doctrine$migrations.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-ocramius$proxy-manager.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-zendframework$zend-code.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-ocramius$package-versions.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-zendframework$zend-eventmanager.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-squizlabs$php-codesniffer.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-zendframework$zend-stdlib.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-athletic$athletic.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-zendframework$zend-hydrator.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-phpdocumentor$reflection.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$browser-kit.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$css-selector.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$debug-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$dom-crawler.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$expression-language.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$ldap.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$property-info.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$proxy-manager-bridge.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$security.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$security-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$security-guard.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$security-http.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$serializer.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$twig-bundle.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$validator.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$var-dumper.json from cache
Reading C:/Users/Martixy/AppData/Local/Composer/repo/https---packagist.org/provider-symfony$web-profiler-bundle.json from cache
Resolving dependencies through SAT
Dependency resolution completed in 0.950 seconds
Analyzed 6584 packages to resolve dependencies
Analyzed 215355 rules to resolve dependencies
  - Installing zendframework/zend-eventmanager (3.0.1)
Reading C:/Users/Martixy/AppData/Local/Composer/files/zendframework/zend-eventmanager/fdde12b1b61e015390902a56921ca7295985e060.zip from cache
    Loading from cache
    Extracting archive

    REASON: zendframework/zend-code 2.6.2 requires zendframework/zend-eventmanager ^2.6|^3.0 -> satisfiable by zendframework/zend-eventmanager[2.6.0, 2.6.1, 2.6.2, 2.6.3, 3.0.0, 3.0.1].

  - Installing zendframework/zend-code (2.6.2)
Reading C:/Users/Martixy/AppData/Local/Composer/files/zendframework/zend-code/63056ecceaaaf45cc649bccfcaf32273ba013392.zip from cache
    Loading from cache
    Extracting archive

    REASON: ocramius/proxy-manager 1.0.2 requires zendframework/zend-code >2.2.5,<3.0 -> satisfiable by zendframework/zend-code[2.2.10, 2.2.6, 2.2.7, 2.2.8, 2.2.9, 2.3.0, 2.3.1, 2.3.2, 2.3.3, 2.3.4, 2
.3.5, 2.3.6, 2.3.7, 2.3.8, 2.3.9, 2.4.0, 2.4.1, 2.4.2, 2.4.3, 2.4.4, 2.4.5, 2.4.6, 2.4.7, 2.4.8, 2.4.9, 2.5.0, 2.5.1, 2.5.2, 2.5.3, 2.6.0, 2.6.1, 2.6.2].

  - Installing ocramius/proxy-manager (1.0.2)
Reading C:/Users/Martixy/AppData/Local/Composer/files/ocramius/proxy-manager/40161808d127b2122f3f2fe6c0dc4c4984f64c37.zip from cache
    Loading from cache
    Extracting archive

    REASON: doctrine/migrations 1.4.1 requires ocramius/proxy-manager ^1.0|^2.0 -> satisfiable by ocramius/proxy-manager[1.0.0, 1.0.1, 1.0.2, 2.0.0, 2.0.1].

  - Installing doctrine/migrations (1.4.1)
Reading C:/Users/Martixy/AppData/Local/Composer/files/doctrine/migrations/57a9703f649f2e0fe43deda9ced60977443fabb6.zip from cache
    Loading from cache
    Extracting archive

    REASON: doctrine/doctrine-migrations-bundle v1.1.0 requires doctrine/migrations ~1.0 -> satisfiable by doctrine/migrations[1.4.1, v1.0.0, v1.1.0, v1.2.0, v1.2.1, v1.2.2, v1.3.0, v1.3.1, v1.4.0].

  - Installing doctrine/doctrine-migrations-bundle (v1.1.0)
Reading C:/Users/Martixy/AppData/Local/Composer/files/doctrine/doctrine-migrations-bundle/6b285f2d32dbb95eb404f13c52b0eb9eaa38be72.zip from cache
    Loading from cache
    Extracting archive

    REASON: Required by the root package: Install command rule (install doctrine/doctrine-migrations-bundle v1.1.0)

zendframework/zend-eventmanager suggests installing container-interop/container-interop (^1.1.0, to use the lazy listeners feature)
zendframework/zend-eventmanager suggests installing zendframework/zend-stdlib (^2.7.3 || ^3.0, to use the FilterChain feature)
zendframework/zend-code suggests installing zendframework/zend-stdlib (Zend\Stdlib component)
ocramius/proxy-manager suggests installing zendframework/zend-stdlib (To use the hydrator proxy)
ocramius/proxy-manager suggests installing ocramius/generated-hydrator (To have very fast object to array to object conversion for ghost objects)
ocramius/proxy-manager suggests installing zendframework/zend-xmlrpc (To have the XmlRpc adapter (Remote Object feature))
ocramius/proxy-manager suggests installing zendframework/zend-json (To have the JsonRpc adapter (Remote Object feature))
ocramius/proxy-manager suggests installing zendframework/zend-soap (To have the Soap adapter (Remote Object feature))
Reading D:\WebDev\[...]/vendor/composer/installed.json
Resolving dependencies through SAT
Dependency resolution completed in 0.001 seconds
Writing lock file
Generating autoload files
> post-update-cmd: Incenteev\ParameterHandler\ScriptHandler::buildParameters
Updating the "app/config/parameters.yml" file
> post-update-cmd: Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::buildBootstrap
> post-update-cmd: Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache

 // Clearing the cache for the dev environment with debug true

 [OK] Cache for the "dev" environment (debug=true) was successfully cleared.

> post-update-cmd: Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::installAssets

 Trying to install assets as relative symbolic links.

 --------- ----------------- ------------------
            Bundle            Method / Error
 --------- ----------------- ------------------
  WARNING   FrameworkBundle   absolute symlink
  WARNING   EasyAdminBundle   absolute symlink
 --------- ----------------- ------------------

 [OK] All assets were successfully installed.

> post-update-cmd: Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::installRequirementsFile
> post-update-cmd: Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::prepareDeploymentTarget
```

And I expected this to happen:
Due to the caret, you'd think it would download v1.1.1 (as it did when I ran the same command on the server). But for some reason that's not the case...
Notice how in `composer.json`, the auto-inserted line for that package has no caret.

Furthermore, this is a Win7x64 and
`Composer version 1.0-dev (bc1d92aeee090d17f20548d3902d07d56933f029) 2016-04-03 17:44:22`

**Comment by curry684** [comment]
composer require doctrine/doctrine-migrations-bundle "^1.1" -vvv

The correct syntax is:

    composer require doctrine/doctrine-migrations-bundle:^1.1 -vvv

Right now you're instructing it to require 2 packages: `doctrine/doctrine-migrations-bundle` and `^1.1`. As the first is already required at `1.1` nothing is changed. This explains all your output.

What I don't understand is that it doesn't err in any way on the `^1.1` argument...

**Comment by alcohol** [comment]
Actually, what he ran is also correct:

```
composer require doctrine/doctrine-migrations-bundle "^1.1" -vvv
```

This gets parsed by `Composer\Package\Version\VersionParser::parseNameVersionPairs` into:

```
array(1) {
  [0]=>
  array(2) {
    ["name"]=>
    string(12) "doctrine/doctrine-migrations-bundle"
    ["version"]=>
    string(4) "^1.1"
  }
}
```

What I don't understand is why it didn't update your version requirement in your `composer.json`.

**Comment by curry684** [comment]
Yeah I found out already while debugging hehe.

I took your entire `composer.json` and ran it in various ways, in all cases `^1.1` is inserted and `1.1.1` is intalled, so no clue what's happening here.

**Comment by alcohol** [comment]
Most notable for me was that I see no "./composer.json has been updated" feedback at all. It's like the requirement for 1.1 was already in his `composer.json" before he ran this command. Could it be require does replace existing requirements?

**Comment by curry684** [comment]
I think Windows' batch file processing biting you here:
```sh
C:\>echo "^1.1"
"^1.1"

C:\>echo ^1.1
1.1

C:\>echo ^^1.1
^1.1
```
The caret is [somewhat magic](http://www.psteiner.com/2008/12/using-caret-in-windows-batch-files.html). I cannot reproduce the precise issue with any shell on Win10 though.

**Comment by alcohol** [comment]
I did find that too yes. But since he quoted it I was not sure it would be an issue.

**Comment by curry684** [comment]
> Could it be require does replace existing requirements?

It does. No matter if I run it with the requirement at `1.1` before, or absent - in both cases I end up with `^1.1` in the json, and `1.1.1` installed.

**Comment by martixy** [comment]
It isn't a question of updating the `.json`.
That package was never in the json. I included the post-command json(and mentioned so), just because it also updates the json with the caret missing. It was not present before running the command.

@alcohol The `./composer.json has been updated` line is 8th in the output.

Meanwhile I have isolated the problem to specifically powershell. If I run the command from cmd.exe, it works properly.
And after some tinkering, this is the command that worked:
`composer require doctrine/doctrine-migrations-bundle "^^^^1.1"`
(It seems I am not the only person with that [problem](https://stackoverflow.com/questions/9600549/using-a-caret-when-using-git-for-windows-in-powershell))
It seems to be environmental and not a problem with composer.
Thanks for the input.
